tox.ini

```diff
@@ -80,33 +80,37 @@ deps =
 setenv =
   TEST_JUJU_CHANNEL = 2.9/stable
 changedir = {toxinidir}/tests/functional
-commands = functest-run-suite
+commands = functest-run-suite {posargs:--keep-faulty-model}
 deps =
   git+https://github.com/openstack-charmers/zaza.git#egg=zaza
   -r {toxinidir}/tests/functional/requirements.txt
 
-[testenv:func33-jammy]
+[testenv:func31]
 setenv =
   TEST_JUJU_CHANNEL = 3.3/stable
 changedir = {toxinidir}/tests/functional
 commands =
-  # NOTE(mertkirpici): functest-run-suite -b jammy focal
-  # or functest-run-suite -b jammy -b focal does not work
-  # https://github.com/openstack-charmers/zaza/issues/614
-  functest-run-suite -b jammy
+  functest-run-suite {posargs:--keep-faulty-model}
 deps =
-  git+https://github.com/openstack-charmers/zaza.git@libjuju-3.1#egg=zaza[juju-33]
+  git+https://github.com/openstack-charmers/zaza.git@libjuju-3.1#egg=zaza[juju-31]
+  -r {toxinidir}/tests/functional/requirements.txt
+
+[testenv:func32]
```

Comment:
I really do not understand need of these tox environment.

Reply:
Just use different version of zaza for testing.

Reply:
We can simply achieve same by passing proper env and using single tox env. We will and tox  env for every version of Juju?
BTW, why we are adding support for already old Juju 3.2, should we do only 3.4?

Reply:
The newest commit already update to single tox env.

> BTW, why we are adding support for already old Juju 3.2, should we do only 3.4?

The plan is to test all the 3.x versions.

Reply:
Please check any repo I did support for Juju 3.4.